activerecord/lib/active_record/database_configurations.rb

```diff
@@ -104,18 +104,28 @@ def build_configs(configs)
         return configs.configurations if configs.is_a?(DatabaseConfigurations)
         return configs if configs.is_a?(Array)
 
-        build_db_config = configs.each_pair.flat_map do |env_name, config|
-          walk_configs(env_name.to_s, "primary", config)
-        end.flatten.compact
+        db_configs = configs.flat_map do |env_name, config|
+          if config.is_a?(Hash) && config.all? { |k, v| v.is_a?(Hash) }
```

Comment:
Minor but tomorrow when we pair on this let's make sure to underscore the `k` since it's not used.